include/libopencm3/stm32/h7/rcc.h

```diff
@@ -429,9 +781,64 @@ enum rcc_periph_rst {
 };
 
 #undef _REG_BIT
+/**@}*/
+
 
+/** @defgroup rcc_file RCC peripheral API
+ *
+ * @ingroup peripheral_apis
+ */
 #include <libopencm3/stm32/common/rcc_common_all.h>
 
+BEGIN_DECLS
+
+/**
+ * Setup the base PLLs and clock domains for the STM32H7. This function will
+ * utilize the users input parameters to configure all 3 PLLs, as well as the
+ * core clock domains (namely SYSCLK, CPU, HCLK, AHB, PCLK1-4) with the
+ * specified dividers. Given the dividers, the RCC module will track the
+ * the configured frequency for each of these clock domains.
+ *
+ *   Note: If clock sources, configs, divider, etc. are modified outside of
+ *         this module, the frequency tracking may be undefined.
+ *   Note: Clock tree is fairly complex, see RM0433 Section 7
+ *         for details.
+ * @param[in] config  Input config structure defining desired setup.
+ */
+void rcc_clock_setup_pll(const struct rcc_pll_config *config);
+
+/**
+ * Get the clock rate (in hz) of the specified clock source. There are
+ * numerous clock sources and configurations on the H7, so rates for each
+ * configured peripheral clock are aimed to be discoverd/calculated by this
+ * module such that the user does not need to know how the MCU is configured
+ * in order to utilize a peripheral clock.
+ * @param[in] source  Clock source desired to be fetched.
+ * @return Clock rate in Hz for the specified clock. 0 if undefined or error.
+ */
```

Comment:
would suggest, if you _really_ see how this is important, to make the name be get_clockrate or clockspeed, or something, to make it clear it's not getting the clock source

Reply:
how about `get_clock_freq()`